In [15]:
numeric_df = dataset.select_dtypes(include=['int64', 'float64'])

In [16]:
numeric_df.corr()

year   mileage  engine_hp  owner_count  vehicle_age  \
year              1.000000 -0.782824  -0.001647    -0.648160    -0.998914   
mileage          -0.782824  1.000000  -0.001158     0.513431     0.782042   
engine_hp        -0.001647 -0.001158   1.000000    -0.004388     0.001457   
owner_count      -0.648160  0.513431  -0.004388     1.000000     0.645281   
vehicle_age      -0.998914  0.782042   0.001457     0.645281     1.000000   
mileage_per_year -0.115253  0.604157  -0.002433     0.094131     0.099930   
brand_popularity -0.005493  0.003359   0.081157    -0.000119     0.005747   
price             0.663266 -0.617453   0.654184    -0.452540    -0.661557   

                  mileage_per_year  brand_popularity     price  
year                     -0.115253         -0.005493  0.663266  
mileage                   0.604157          0.003359 -0.617453  
engine_hp                -0.002433          0.081157  0.654184  
owner_count               0.094131         -0.000119 -0.452540  
vehicle_age               0.099930          0.005747 -0.661557  
mileage_per_year          1.000000         -0.000332 -0.216261  
brand_popularity         -0.000332          1.000000  0.056355  
price                    -0.216261          0.056355  1.000000

# What is the relation between price and Vehicle Age 

Vehicle age shows a strong negative correlation (-0.66) with price, indicating that as cars get older, their market value decreases significantly.

# What is the relation between price and Engine_hp

The correlation between engine horsepower (engine_hp) and price is 0.654184, indicating a strong positive correlation.
This means that cars with higher horsepower generally have higher prices, as increased engine power is often associated with better performance, premium features, and higher-end models.

In [17]:
numeric_df.cov()

year       mileage      engine_hp   owner_count  \
year                  15.182354 -2.193221e+05      -0.600832 -3.901281e+00   
mileage          -219322.148802  5.170082e+09   -7796.967274  5.702773e+04   
engine_hp             -0.600832 -7.796967e+03    8764.045678 -6.345823e-01   
owner_count           -3.901281  5.702773e+04      -0.634582  2.386222e+00   
vehicle_age          -14.887997  2.150888e+05       0.521727  3.812790e+00   
mileage_per_year   -2745.978195  2.656292e+08   -1392.863771  8.891277e+02   
brand_popularity      -0.000005  5.667725e-02       0.001783 -4.307409e-08   
price              35294.671211 -6.063236e+08  836380.033775 -9.546948e+03   

                    vehicle_age  mileage_per_year  brand_popularity  \
year                 -14.887997     -2.745978e+03     -5.022443e-06   
mileage           215088.804564      2.656292e+08      5.667725e-02   
engine_hp              0.521727     -1.392864e+03      1.782880e-03   
owner_count            3.812790      8.891277e+02     -4.307409e-08   
vehicle_age           14.631122      2.337296e+03      5.158589e-06   
mileage_per_year    2337.296281      3.738988e+07     -4.759557e-04   
brand_popularity       0.000005     -4.759557e-04      5.506671e-08   
price             -34558.707600     -1.805952e+07      1.806058e-01   

                         price  
year              3.529467e+04  
mileage          -6.063236e+08  
engine_hp         8.363800e+05  
owner_count      -9.546948e+03  
vehicle_age      -3.455871e+04  
mileage_per_year -1.805952e+07  
brand_popularity  1.806058e-01  
price             1.865106e+08

# What does the covariance between Year and Price indicate about their relationship?

The covariance between Year and Price is 35,294.67, which is positive, indicating that as the year of the car increases (newer models), the price also tends to increase.

In [3]:
df=dataset.copy()

# Is there a significant difference in car prices between Electric and Diesel vehicles?

In [25]:
from scipy.stats import ttest_ind
group1 = df[df['fuel_type'] == 'Electric']['price']
group2 = df[df['fuel_type'] == 'Diesel']['price']

ttest_ind(group1, group2)

TtestResult(statistic=10.564662552429668, pvalue=4.665994462103537e-26, df=44441.0)

Since the p-value is far less than 0.05, the difference in average prices between Electric and Diesel cars is statistically significant.
This means that Electric vehicles tend to have significantly higher prices than Diesel vehicles, likely due to newer technology, higher production costs, and premium positioning in the market.